### PR TITLE
Create Search_Leaders.dart

### PR DESCRIPTION
**Pull request by RimjhimGupta**
An element is leader if it is greater than all the elements to its right side. Also, the rightmost element is always a leader.
This pull request is for Hacktober Fest 2020.

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: search/Search_Leaders.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+void printLeaders(List<int> arr, int size) {
+  for (int i = 0; i < size; i++) {
+    int j;
+    for (j = i + 1; j < size; j++) {
+      if (arr[i] < arr[j]) break;
+    }
+    if (j == size) // the loop didn't break
```
Comment (akashgkrishnan) [inline search/Search_Leaders.dart:7]: We can use a Stack Data Structure to solve this problem which will reduce the time complexity from O(n^2)
to O(N).
in the first pass, we can find the next greater element using stack,
and in next pass we can return the elements in array

File: search/Search_Leaders.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      print(" ");
+    }
+  }
+}
+
+/* Driver program to test above function */
+void main() {
+  List<int> arr = [22, 37, 4, 1, 18, 10];
+  int n = arr.length;
+  printLeaders(arr, n);
```
Comment (akashgkrishnan) [inline search/Search_Leaders.dart:19]: Add test cases

**Review by akashgkrishnan — CHANGES_REQUESTED** [review]

**Comment by RimjhimGupta** [comment]
@Parowicz Sir, I request you to please merge my pull request for Hacktober Fest 2020.

**Comment by Parowicz** [comment]
Can you add test cases and answer issues pointed by @akashgkrishnan?

**Comment by RimjhimGupta** [comment]
@Parowicz Sir, I am a beginner, kindly help me in resolving the same.
Thank you

**Comment by akashgkrishnan** [comment]
> @Parowicz Sir, I am a beginner, kindly help me in resolving the same.
> Thank you

@RimjhimGupta can you add the test cases as suggested by @Parowicz ? I'll suggest you look into other files in the repo and find how tests are implemented in dart and also add test cases covering major cases and edge cases
also, try to implement a more efficient algorithm as the time complexity seems to be high and there are more efficient approaches. Happy coding

**Comment by akashgk** [comment]
HI @RimjhimGupta 

Would you like to complete this PR?
